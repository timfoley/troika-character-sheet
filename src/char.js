export default {
	name: "Dahddy",
	background: "Beef Steak",
	special: "",
	skill: 5,
	stamina: {
		max: 18,
		remaining: 18
	},
	luck: {
		max: 10,
		remaining: 10
	},
	skills: [
		{
			name: "Hammer Fighting",
			rank: 3
		},
		{
			name: "Fist Fighting",
			rank: 2
		},
		{
			name: "Wrestle",
			rank: 2
		},
		{
			name: "Swim",
			rank: 2
		},
		{
			name: "Climb",
			rank: 2
		},
		{
			name: "Run",
			rank: 2
		},
		{
			name: "Grill",
			rank: 2
		}
	],
	weapons: [
		{
			name: "Steak Knife",
			damage: [2, 2, 2, 2, 4, 8, 10],
			twoHanded: false,
			armorPiercing: false
		},
		{
			name: "Meat Tenderizer",
			damage: [1, 2, 4, 6, 8, 10, 12],
			twoHanded: false,
			armorPiercing: true
		}
	],
	inventory: [
		{
			name: "Hammer",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Knife",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Portable grill",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Steaks",
			notes: "premium quality",
			quantity: 7,
			slots: 1
		},
		{
			name: "Flask of oil",
			notes: "",
			quantity: 1,
			slots: 1
		},
		{
			name: "Lantern",
			notes: "",
			quantity: 1,
			slots: 1
		}
	],
	wearing: "Waistcoat",
	money: 12,
	provisions: 6
};
