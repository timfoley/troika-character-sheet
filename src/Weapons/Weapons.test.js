import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";

import { Weapons } from "./index.js";

import char from "../char";

it("renders Weapons without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(
		<Weapons updateCharacter={() => {}} weapons={char.weapons} />,
		div
	);
	ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
	const { container } = render(
		<Weapons updateCharacter={() => {}} weapons={char.weapons} />
	);
	expect(container.firstChild).toMatchSnapshot();
});

// it("renders a heading called 'Weapons'", () => {
//     const div = document.createElement("div");
//     ReactDOM.render(<Weapons />, div);
//     const { getByLabelText } = getQueriesForElement(div);
//     const input = getByLabelText(/name/i);
//     expect(input).toHaveAttribute("type", "text");
// })
