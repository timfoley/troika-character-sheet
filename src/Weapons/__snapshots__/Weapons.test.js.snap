// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="weapons"
>
  <h2>
    Weapons
  </h2>
  <div
    class="weapons__grid"
  >
    <div
      class="weapon__name"
    >
      <input
        aria-label="weapon name"
        class="weapon__name__input"
        type="text"
        value="Steak Knife"
      />
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="1 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="2"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        1
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="2 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="2"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        2
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="3 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="2"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        3
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="4 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="2"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        4
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="5 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="4"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        5
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="6 damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="8"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        6
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="7+ damage value for Steak Knife"
        class="weapon__damage__input weapon__damage__input--0 hide-spinners"
        type="number"
        value="10"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        7+
      </div>
    </div>
    <div
      class="weapon__name"
    >
      <input
        aria-label="weapon name"
        class="weapon__name__input"
        type="text"
        value="Meat Tenderizer"
      />
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="1 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="1"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        1
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="2 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="2"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        2
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="3 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="4"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        3
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="4 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="6"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        4
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="5 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="8"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        5
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="6 damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="10"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        6
      </div>
    </div>
    <div
      class="weapon__damage"
    >
      <input
        aria-label="7+ damage value for Meat Tenderizer"
        class="weapon__damage__input weapon__damage__input--1 hide-spinners"
        type="number"
        value="12"
      />
      <div
        aria-hidden="true"
        class="weapon__damage__roll"
      >
        7+
      </div>
    </div>
  </div>
  <div
    class="weapons__add-button clickable"
    tabindex="0"
  >
    Add Weapon
  </div>
</div>
`;
